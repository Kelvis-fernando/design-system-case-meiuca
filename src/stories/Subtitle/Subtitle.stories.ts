import type { Meta, StoryObj } from "@storybook/react";

import { Subtitle } from "./Subtitle";

const meta = {
  title: "Components/Subtitle",
  component: Subtitle,
  tags: ["autodocs"],
} satisfies Meta<typeof Subtitle>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    text: "This is the First Subtitle",
    size: "lg",
  },
};
