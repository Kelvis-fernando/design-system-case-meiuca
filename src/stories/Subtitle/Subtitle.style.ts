import tokens from "../../../tokens/global.json";

export const getSubtitleStyle = (size: string) => {
  const subtitleSize = () => {
    switch (size) {
      case "sm":
        return tokens.font.size.xxxs.value;
      case "md":
        return tokens.font.size.xxs.value;
      case "lg":
        return tokens.font.size.xs.value;
    }
  };
  return {
    color: tokens.neutral.color[3].value,
    fontSize: subtitleSize(),
    fontFamily: tokens.font.family.highlight.value,
  };
};
