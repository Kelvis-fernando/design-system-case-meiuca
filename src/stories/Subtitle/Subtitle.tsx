import React from "react";
import { getSubtitleStyle } from "./Subtitle.style";

export interface HeadingProps {
  text: string;
  size?: "sm" | "md" | "lg";
}

export const Subtitle: React.FC<HeadingProps> = ({ text, size }) => {
  return (
    <p style={getSubtitleStyle(size !== undefined ? size : "md")}>{text}</p>
  );
};
