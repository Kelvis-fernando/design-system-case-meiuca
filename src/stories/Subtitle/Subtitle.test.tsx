import { render, screen } from "@testing-library/react";
import { Subtitle } from "./Subtitle";

test("should render Subtitle", () => {
  const { container } = render(<Subtitle size="md" text={"Subtitle"} />);
  const linkElement = screen.getByText(/Subtitle/i);

  expect(linkElement).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
