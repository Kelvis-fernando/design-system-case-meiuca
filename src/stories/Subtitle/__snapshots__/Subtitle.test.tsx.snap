// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Subtitle 1`] = `
<div>
  <p
    style="color: rgb(166, 170, 175); font-size: 14px; font-family: 'Archivo';"
  >
    Subtitle
  </p>
</div>
`;

exports[`should render heading 1`] = `
<div>
  <p
    style="color: rgb(166, 170, 175); font-family: 'Archivo';"
  >
    Heading
  </p>
</div>
`;
